spring-mvc/src/main/java/com/kodemon/springmvc/controllers/UserController.java

```diff
@@ -223,4 +226,52 @@ public String logout(ServletRequest r, Model model, Locale locale) {
     private String getMessage(String code, Locale locale, Object... args) {
         return messageSource.getMessage(code, args, locale);
     }
+
+    /**
+     * Page for selecting first 6 pokemons to be used for fight
+     *
+     * @param r      servlet request to get logged user
+     * @param model  data to display
+     * @param locale locale settings
+     * @return JSP page name
+     */
+    @RequestMapping(value = "/reorder", method = RequestMethod.GET)
+    public String reorder(ServletRequest r, Model model, Locale locale) {
+        HttpServletRequest request = (HttpServletRequest) r;
+        HttpSession session = request.getSession();
+        if (session.getAttribute("authenticatedUser") == null) {
+            model.addAttribute("alert_warning", getMessage("warning.user.notLoggedIn", locale));
+            return "login";
+        }
+        UserDTO currentUser = (UserDTO) session.getAttribute("authenticatedUser");
+        model.addAttribute("pokemons", currentUser.getPokemons());
+        model.addAttribute("activePokemons", currentUser.getActivePokemons());
+        model.addAttribute("numberOfPokemonsForFight", MAX_ACTIVE_POKEMON);
+        return "user/reorder";
+    }
+
+    @RequestMapping(value = "setFirstSixPokemons", method = RequestMethod.POST)
+    public String setFirstSixPokemons(ServletRequest r, Model model, Locale locale) {
+        HttpServletRequest request = (HttpServletRequest) r;
+        HttpSession session = request.getSession();
+
+        if (session.getAttribute("authenticatedUser") == null) {
+            model.addAttribute("alert_warning", getMessage("warning.user.notLoggedIn", locale));
+            return "login";
+        }
+
+        List<Integer> pokemonIndices = new ArrayList<>();
+        for(String i : request.getParameterValues("pokemon")) {
+            pokemonIndices.add(Integer.valueOf(i));
+        }
+
+        UserDTO currentUser = (UserDTO) session.getAttribute("authenticatedUser");
+        userFacade.chooseActivePokemons(currentUser.getId(), pokemonIndices);
+
+        model.addAttribute("alert_success", getMessage("success.reorder.changed", locale));
+        UserDTO user = userFacade.findUserByUserNameIgnoringCaseIncludeSubstrings(currentUser.getUserName()).iterator().next();
+        model.addAttribute("trainer", user);
+
+        return "user/detail";
```

Comment:
Here, please use `return redirect:/user/detail` in order to change the URL in the address bar. Plus, add `RedirectParameters redirectParameters` as the parameter for this method and instead of `model.addAttribute()` use `redirectAttributes.addFlashAttribute(...)` in the lines 271, 273. This will cause app to redirect, whereas just return `user/detail` causes only the change of the view. Probably this should be done on line 260, but you can leave it as is.